ml-prims/src/random/rng.h

```diff
@@ -200,15 +198,31 @@ class Rng {
    * @param start start of the range
    * @param end end of the range
    * @param stream stream where to launch the kernel
+   * @{
    */
+  template <typename Type>
   void uniform(Type *ptr, int len, Type start, Type end,
                cudaStream_t stream = 0) {
+    static_assert(std::is_floating_point<Type>::value,
+                  "Type for 'uniform' can only be floating point type!");
     randImpl(offset, ptr, len,
              [=] __device__(Type val, unsigned idx) {
-               return (end - start) * val + start;
+               return (val * (end - start)) + start;
+             },
+             NumThreads, nBlocks, type, stream);
+  }
+  template <typename IntType>
+  void uniformInt(IntType *ptr, int len, IntType start, IntType end,
```

Comment:
what about 
```
template <typename IntType, LenType, RangeType>
void uniformInt(IntType *ptr, LenType len, RangeType start, RangeType end,
                                                         cudaStream_t stream = 0) {
```

to allow for uint length and signed range values.

Reply:
Thanks Levi. That's a good point! I have added another template param to choose the type for hte length of the arrays itself.

Reply:
Actually, the `RangeType` can be the same as `IntType`

Reply:
ah.. wait.. I see your point.
It's about generating random signed integers as well, right?
Makes sense. I'll just have to update the *Generator classes for this.
Let me do that.

Reply:
have added support for generating signed int's too. thanks for pointing this out.